src/ric/2C4C4.c

```diff
@@ -686,54 +686,53 @@ void func_80169C10(Entity* entity) {
 }
 
 // made by blueprint #5, see step 0 of EntityCrossBoomerang
-void EntityCrossShadow(Entity* entity) {
-    Point16* temp;
-    s16* ptr;
+void EntityCrossShadow(Entity* self) {
+    s16* temp;
 
-    switch (entity->step) {
+    switch (self->step) {
     case 0:
-        entity->flags = FLAG_UNK_04000000 | FLAG_UNK_08000000;
+        self->flags = FLAG_UNK_04000000 | FLAG_UNK_08000000;
         // the parent pointer is set in RicEntityEntFactory.
         // the value of unk84 is set in EntityCrossBoomerang
-        entity->ext.crossBoomerang.unk84 =
-            entity->ext.factory.parent->ext.crossBoomerang.unk84;
-        entity->animSet = ANIMSET_OVL(17);
-        entity->animCurFrame = D_80155E68[entity->params];
-        entity->unk5A = 0x66;
-        entity->palette = 0x81B0;
-        entity->drawMode = DRAW_TPAGE;
-        entity->facingLeft = PLAYER.facingLeft;
-        entity->zPriority = PLAYER.zPriority;
-        entity->drawFlags = FLAG_DRAW_ROTZ;
-        entity->rotZ = 0xC00;
-        entity->step++;
+        self->ext.crossBoomerang.unk84 =
+            self->ext.factory.parent->ext.crossBoomerang.unk84;
+        self->animSet = ANIMSET_OVL(17);
+        self->animCurFrame = D_80155E68[self->params];
+        self->unk5A = 0x66;
+        self->palette = 0x81B0;
+        self->drawMode = DRAW_TPAGE;
+        self->facingLeft = PLAYER.facingLeft;
+        self->zPriority = PLAYER.zPriority;
+        self->drawFlags = FLAG_DRAW_ROTZ;
+        self->rotZ = 0xC00;
+        self->step++;
         break;
 
     case 1:
-        entity->rotZ -= 0x80;
-        if (entity->ext.factory.parent->step == 7) {
-            entity->step++;
-            entity->ext.generic.unk7C.s = (entity->params + 1) * 4;
+        self->rotZ -= 0x80;
+        if (self->ext.factory.parent->step == 7) {
+            self->step++;
+            self->ext.timer.t = (self->params + 1) * 4;
```

Comment:
I think it doesn't make any difference in this case but we're using multiple ET structs for the same entity

Reply:
Good point, changed to use cross boomerang's timer